### PR TITLE
Server/shadow: Fix issue 2796. Mouse cursor draw back issue

### PR DESCRIPTION
**Pull request by realjiangms**
Fix the issue we discussed in #2796

### CODE REVIEW

File: server/shadow/X11/x11_shadow.c [hunk 1]
```diff
@@ -380,7 +386,26 @@ int x11_shadow_pointer_position_update(x11ShadowSubsystem* subsystem)
...
 
-	return shadow_client_boardcast_msg(subsystem->server, NULL, msgId, (SHADOW_MSG_OUT*) msg, NULL) ? 1 : -1;
+	server = subsystem->server;
+
+	ArrayList_Lock(server->clients);
+	for (index = 0; index < ArrayList_Count(server->clients); index++)
+	{
+		client = (rdpShadowClient*)ArrayList_GetItem(server->clients, index);
+		if (client == subsystem->lastMouseClient)
+		{
```
Comment (bmiklautz) [inline server/shadow/X11/x11_shadow.c]: Coding style. Only line argument shouldn't have {}

File: server/shadow/X11/x11_shadow.c [hunk 2]
```diff
@@ -380,7 +386,26 @@ int x11_shadow_pointer_position_update(x11ShadowSubsystem* subsystem)
...
+			/* Skip the client which send us the latest mouse event */
+			continue;
+		}
+
+		if (shadow_client_post_msg(client, NULL, msgId, (SHADOW_MSG_OUT*) msg, NULL))
+		{
+			count++;
+		}
```
Comment (bmiklautz) [inline server/shadow/X11/x11_shadow.c]: Coding style. Only line argument shouldn't have {}

**Comment by bmiklautz** [comment]
Except the coding style it looks good to me \o/.

**Comment by realjiangms** [comment]
I understand ^_^

But can I argue that always have {} would makes it have less line difference when someone need to add more lines of code in the 'if' block, and also makes merge easier in Parallel Development.

I would accept the comment as default behavior if there's no one say 'please keep the current style' in 24hrs ^_^, and change the code to following:

```
if (client == subsystem->lastMouseClient)
	continue; /* Skip the client which send us the latest mouse event */


if (shadow_client_post_msg(client, NULL, msgId, (SHADOW_MSG_OUT*) msg, NULL))
	count++;
```

**Comment by bmiklautz** [comment]
@realjiangms all you say is tru. I totally agree with you and personally also prefer to always have {} blocks, even for single lines. However the "unwritten" coding guide for FreeRDP says that single lines should not have {}. If in doubt there is a astyle script in scripts/format_code.sh.

**Comment by realjiangms** [comment]
Then a commonly accepted rule wins, even it is 'unwritten'.

**Comment by realjiangms** [comment]
Having some trouble with astyle...

**Comment by bmiklautz** [comment]
@realjiangms it's of course not required to use the script.

**Comment by realjiangms** [comment]
git push failed. Let me try again

**Comment by bmiklautz** [comment]
Perfect. Thanks a lot.
